scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateRequest.java

```diff
@@ -444,8 +444,8 @@ boolean isSupportedKeyType(X509Certificate cert) {
 
 		for (ClientCertificateType type : certificateTypes) {
 			boolean isCompatibleType = type.isCompatibleWithKeyAlgorithm(cert.getPublicKey().getAlgorithm());
-			boolean meetsSigningRequirements = !type.requiresSigningCapability() ||
-					(type.requiresSigningCapability() && cert.getKeyUsage() != null && cert.getKeyUsage()[0]);
+			boolean meetsSigningRequirements = !type.requiresSigningCapability()
+					|| (type.requiresSigningCapability() && (cert.getKeyUsage() == null || cert.getKeyUsage()[0]));
```

Comment:
!a || (a && b) := !a || b

Could this be reduced to the logic equivalent?

Reply:
And please add a comment, that it's appreciated, if someone can provide details about the usage of the "key usage" and "extended key usage" related to a "certificate request".